### DISCUSSION TITLE
Creating binary mask file

### DISCUSSION THREAD

**Discussion by epehlivanli** [insarlab/MintPy#1032] [Q&A]
Hi everyone!

I hope everything is all right! I am dealing with applying a mask to extract the area of interest (main railroad). I tried in QGIS, I used an imported railroad with QuickOSM. However, I was only able to apply it to raster file as a tiff. I also tried [generate_mask.py](https://github.com/insarlab/MintPy/blob/main/src/mintpy/generate_mask.py) to convert binary mask GeoTiff to.h5. It also did not work. 
Is there any chance to convert .h5 to vector formats or do I have to convert binary mask in HDF5 format with containing same information as the file I want to apply mask? 

Any advice or reference is appreciated!

**Comment by yunjunz** [comment]
I have not tried to convert a raster image into vectors, but I am not a heavy GIS user, would love to hear some examples if anyone have.

To mask raster images in mintpy using `mask.py`, the mask file and data file has to be in the same size, and usually, the mask file is a 2D matrix in bool type, with the masked-out pixels in 0 in the mask file. Run `mask.py -h` to check its usage please.

**Comment by falkamelung** [comment]
To follow up on this, we have a *geoTiff file saved with QGIS. We need to convert this file to a *.h5 file. Gdal converts to HDF4  but not to HDF5. So the question is simply how to convert from geoTiff to HDF5.
Thank you!

**Reply by yunjunz** [reply]
mintpy could now read single-band geotiff files, so I believe you could use it directly in `mask.py`.

**Comment by falkamelung** [comment]
mask.py works perfectly using the geoTiff file. Thank you! 

Only the geoTiff file has {0,1} whereas the mask files are Boolean {True, False} and therefor `add.py` does not work. Have to figure this out.

```
(Pdb++) n
[4] > /home/exouser/operations/rsmas_insar/tools/MintPy/src/mintpy/add.py(123)add_file()
data = add_matrix(data, data2)
(Pdb++) data
array([[0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       ...,
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.]], dtype=float32)
(Pdb++) data2
array([[ True,  True,  True, ..., False, False, False],
       [ True,  True,  True, ..., False, False, False],
       [ True, False, False, ..., False, False, False],
       ...,
       [ True,  True,  True, ..., False, False, False],
       [False,  True,  True, ..., False, False, False],
       [ True,  True,  True, ..., False, False, False]])
```

**Reply by yunjunz** [reply]
I think it should work, as we convert everything to float32 before applying the sum operation. https://github.com/insarlab/MintPy/blob/5f1ae8d3cb4a015c2931255d581e78fd4bfdc061/src/mintpy/add.py#L20-L21
